src/main/java/hudson/plugins/throttleconcurrents/ThrottleQueueTaskDispatcher.java

```diff
@@ -228,7 +228,7 @@ private int buildsOfProjectOnAllNodes(Task task) {
     private int buildsOnExecutor(Task task, Executor exec) {
         int runCount = 0;
         if (exec.getCurrentExecutable() != null
-            && exec.getCurrentExecutable().getParent() == task) {
+            && task.equals(exec.getCurrentExecutable().getParent())) {
```

Comment:
A couple of reasons:
1. Our job class has equals/hashCode overriden
2. In Jenkins core, they use equals for similar purposes, so I assume, that it is OK. E.g. here's a chunk of code from the hudson.model.Queue class:

    public boolean cancel(Task p) {
        lock.lock();
        try { try {
            LOGGER.log(Level.FINE, "Cancelling {0}", p);
            for (WaitingItem item : waitingList) {
                if (item.task.equals(p)) {
                    return item.cancel(this);
                }
            }
            // use bitwise-OR to make sure that both branches get evaluated all the time
            return blockedProjects.cancel(p) != null | buildables.cancel(p) != null;
        } finally { updateSnapshot(); } } finally {
            lock.unlock();
        }
    }

3. It's generally considered a good practice to use equals() instead of reference equality checks.